### PR TITLE
Introduce hidden_dim_grid for using a separate latent dim up to mesh

### PR DESCRIPTION
**Pull request by joeloskarsson**
Now also uses hidden_dim_grid for g2m and m2g, then projects up for processor.

Don't merge yet ;)

### CODE REVIEW

File: neural_lam/models/base_graph_model.py [hunk 1]
```diff
@@ -83,37 +95,50 @@ def __init__(
...
                 self.boundary_embedder = utils.make_mlp(
-                    [self.boundary_dim] + self.mlp_blueprint_end
+                    [self.boundary_dim] + self.grid_mlp_blueprint_end
                 )
 
-        self.g2m_embedder = utils.make_mlp([g2m_dim] + self.mlp_blueprint_end)
-        self.m2g_embedder = utils.make_mlp([m2g_dim] + self.mlp_blueprint_end)
+        # Projections between grid dim and hidden dim before and after processor
+        self.pre_mesh_proj = nn.Sequential(
+            nn.SiLU(), nn.Linear(hidden_dim_grid, args.hidden_dim)
```
Comment (sadamov) [inline neural_lam/models/base_graph_model.py:103]: is there a specific reasoning for SiLU here?

**Review by sadamov — APPROVED** [review]
Looks good to me, was double checking the indexing which is now treating the first and last sub-module with a different hidden dim. thank you!

File: neural_lam/models/base_graph_model.py (hunk 1 above)
Comment (joeloskarsson) [reply neural_lam/models/base_graph_model.py:103]: Just same as in the MLPs https://github.com/joeloskarsson/neural-lam-dev/blob/9468fbb54a3292d6f2058c8da7aedbec5eeaa3cf/neural_lam/utils.py#L271

Comment (joeloskarsson) [reply neural_lam/models/base_graph_model.py:103]: And included an activation function to not have linear layer after linear layer, as that is just unnecessary computation without adding any model flexibility
